Makefile

```diff
@@ -585,37 +586,12 @@ test-%: check-whitespace $(JULIA_BUILD_MODE)
 # download target for some hardcoded windows dependencies
 .PHONY: win-extras wine_path
 win-extras:
-	[ -d $(JULIAHOME)/dist-extras ] || mkdir $(JULIAHOME)/dist-extras
-ifneq ($(BUILD_OS),WINNT)
-ifeq (,$(findstring CYGWIN,$(BUILD_OS)))
-	cp /usr/lib/p7zip/7z /usr/lib/p7zip/7z.so $(JULIAHOME)/dist-extras
-endif
-endif
-ifneq (,$(filter $(ARCH), i386 i486 i586 i686))
-	cd $(JULIAHOME)/dist-extras && \
-	$(JLDOWNLOAD) https://sourceforge.net/projects/sevenzip/files/7-Zip/19.00/7z1900.exe && \
-	$(JLCHECKSUM) 7z1900.exe && \
-	7z x -y 7z1900.exe 7z.exe 7z.dll && \
-	../contrib/windows/winrpm.sh http://download.opensuse.org/repositories/windows:/mingw:/win32/openSUSE_Leap_42.2 \
-		"mingw32-libexpat1 mingw32-zlib1" && \
-	cp usr/i686-w64-mingw32/sys-root/mingw/bin/*.dll .
-else ifeq ($(ARCH),x86_64)
-	cd $(JULIAHOME)/dist-extras && \
-	$(JLDOWNLOAD) https://downloads.sourceforge.net/project/sevenzip/7-Zip/19.00/7z1900-x64.exe && \
-	$(JLCHECKSUM) 7z1900-x64.exe && \
-	7z x -y 7z1900-x64.exe 7z.exe 7z.dll && \
-	../contrib/windows/winrpm.sh http://download.opensuse.org/repositories/windows:/mingw:/win64/openSUSE_Leap_42.2 \
-		"mingw64-libexpat1 mingw64-zlib1" && \
-	cp usr/x86_64-w64-mingw32/sys-root/mingw/bin/*.dll .
-else
-	$(error no win-extras target for ARCH=$(ARCH))
-endif
+	@$(MAKE) -C $(BUILDROOT)/deps install-p7zip
```

Comment:
this recursive calls to MAKE is race-y, so it's generally a bad idea—but also unnecessary (since the `julia-deps` target already does it)?

Reply:
`win-extras` does not depend on `julia-deps`, nor do we want it to.  We don't want to e.g. install LLVM when you run `win-extras`, we only want to install `p7zip`; unfortunately we don't have fine-grained dependency control, so the next best thing is recursive make.  `install-p7zip` is a simple enough operation that I think it's not that bad.  `julia-deps` itself is just a (much larger) recursive make call anyway.

Reply:
eh, ok, it's wrong but not too bad as you say

Reply:
I think the only way this could be done "right" is if we were to include `deps/Makefile` from `Makefile`, giving us access to all those targets.